Widgets/RoomView.h

```diff
@@ -14,6 +17,57 @@ class InstanceSortFilterProxyModel : public QSortFilterProxyModel {
   bool lessThan(const QModelIndex &left, const QModelIndex &right) const override;
 };
 
+struct Proxy {
+  int row;
+  Proxy(int row): row(row) {}
+  virtual int x1() const = 0;
+  virtual int x2() const = 0;
+  virtual int y1() const = 0;
+  virtual int y2() const = 0;
+  int xmin() const { return x1() < x2() ? x1() : x2(); }
+  int xmax() const { return x1() > x2() ? x1() : x2(); }
+  int ymin() const { return y1() < y2() ? y1() : y2(); }
+  int ymax() const { return y1() > y2() ? y1() : y2(); }
+  friend inline bool operator==(Proxy const& lhs, Proxy const& rhs)
+  {
+    return lhs.row == rhs.row;
+  }
+};
+
+struct InstanceProxy : Proxy {
+  InstanceSortFilterProxyModel *model;
+  InstanceProxy(InstanceSortFilterProxyModel *model, int row): Proxy(row), model(model) {}
+  int x1() const override { return model->data(model->index(row, Room::Instance::kXFieldNumber)).toInt(); }
+  int y1() const override { return model->data(model->index(row, Room::Instance::kYFieldNumber)).toInt(); }
+  int x2() const override {
+    int w = sprw();
+    double xscale = model->data(model->index(row, Room::Instance::kXscaleFieldNumber)).toDouble();

```

Comment:
This is effected by #122 which is causing the spatial hash to be glitchy unless manually patched over.